src/test/java/com/wavefront/sdk/common/clients/WavefrontClientTest.java

```diff
@@ -55,5 +61,32 @@ public void testGetBatch() {
   private String createString(int size) {
     return new String(new char[size]).replace("\0", "a");
   }
-  
+
+
+  @Test
+  public void testUrlFormatForService() {
+    List<URI> uris = Lists.newArrayList(
+        URI.create("http://127.0.0.1:2878"),
+        URI.create("http://127.0.0.1:2878/"),
+        URI.create("http://127.0.0.1:2878////"),
+        URI.create("http://localhost:2878/report"),
+        URI.create("http://localhost:2878/report/"),
+        URI.create("http://corp.proxies.acme.com:2878/prod/report/"),
+        URI.create("https://domain.wavefront.com"),
+        URI.create("https://domain.wavefront.com/"),
+        URI.create("https://domain.wavefront.com/report/"),
+        URI.create("https://domain.wavefront.com/report")
+    );
+
+    uris.forEach(this::validateURI);
```

Comment:
I think it's better to do an explicit compare of getReportingUrl().toString() result with the actual expected URL ?